### PR TITLE
AMM-802_Update_ABHA_Search_Label

### PR DESCRIPTION
**Pull request by ravishanigarapu**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.  

# Type of change  

- [x] Bug fix  
- [x] New feature  
- [ ] Enhancement  
- [ ] Refactoring  
- [ ] Documentation  
- [ ] Other ( please specify )  

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/com/wipro/fhir/controller/healthID/SearchHealthIDController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	@Operation(summary = "Search ABHA With Health ID")
+	@PostMapping(value = { "/searchByHealthId" })
+	public String searchHealthID(@RequestBody String healthId) throws FHIRException {
+		OutputResponse response = new OutputResponse();
+		try {
+			logger.info("Search ABHA With Health Id request {} : ", healthId);
+			String resp = searchHealthIDService.searchHealthId(healthId);
+			response.setResponse(resp);
+		} catch (Exception e) {
+			response.setError(5000, e.getMessage());
```
Comment (drtechie) [inline src/main/java/com/wipro/fhir/controller/healthID/SearchHealthIDController.java:37]: Can we stop using this error code and follow proper API specs?

File: src/main/java/com/wipro/fhir/controller/healthID/SearchHealthIDController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	@Operation(summary = "Search ABHA With Mobile Number")
+	@PostMapping(value = { "/searchByMobile" })
+	public String searchHealthIDWithMobile(@RequestBody String mobileSearchDTO) throws FHIRException {
+		OutputResponse response = new OutputResponse();
+		try {
+			logger.info("Search ABHA With Mobile request {} : ", mobileSearchDTO);
+			String resp = searchHealthIDService.searchHealthIdWithMobile(mobileSearchDTO);
+			response.setResponse(resp);
+		} catch (Exception e) {
+			response.setError(5000, e.getMessage());
```
Comment (drtechie) [inline src/main/java/com/wipro/fhir/controller/healthID/SearchHealthIDController.java:53]: Same comment as above.

File: src/main/java/com/wipro/fhir/controller/healthID/SearchHealthIDController.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	@CrossOrigin
+	@Operation(summary = "Search ABHA With Mobile Number")
+	@PostMapping(value = { "/searchByMobile" })
+	public String searchHealthIDWithMobile(@RequestBody String mobileSearchDTO) throws FHIRException {
+		OutputResponse response = new OutputResponse();
+		try {
+			logger.info("Search ABHA With Mobile request {} : ", mobileSearchDTO);
+			String resp = searchHealthIDService.searchHealthIdWithMobile(mobileSearchDTO);
+			response.setResponse(resp);
+		} catch (Exception e) {
```
Comment (drtechie) [inline src/main/java/com/wipro/fhir/controller/healthID/SearchHealthIDController.java:52]: When ABHA is not available, does ABDM throw errors?

**Comment by drtechie** [comment]
@ravishanigarapu please update the PR description with what you have changed.

**Comment by ravishanigarapu** [comment]
If ABHA not exists its throwing below error
Unnable to process the current request due to incorrent data entered.

File: src/main/java/com/wipro/fhir/controller/healthID/SearchHealthIDController.java (hunk 1 above)
Comment (ravishanigarapu) [reply src/main/java/com/wipro/fhir/controller/healthID/SearchHealthIDController.java]: Yes i have updated error code to 500

**Comment by ravishanigarapu** [comment]
In this PR i have handled how to restrict duplicate ABHA while create.
And also created two end points to search ABHA by using Abha ID and mobile number

**Review by drtechie — APPROVED** [review]

**Pull request merged**